### PR TITLE
Refactor and fix archive link bug (#30535)

### PR DESCRIPTION
**Pull request by GiteaBot**
Backport #30535 by wxiaoguang

Regression of #29920
Fixes: https://github.com/go-gitea/gitea/issues/30569

Also this is a rewriting to eliminate the remaining jQuery usages from code.

### CODE REVIEW

**Review by silverwind — APPROVED** [review]

**Comment by silverwind** [comment]
@wxiaoguang seems we are missing `applyElemsCallback` on this branch.

**Comment by wxiaoguang** [comment]
I won't spend time on backporting, the 1.22 branch is broken in my mind.

File: web_src/js/features/repo-common.js [hunk 1]
```diff
@@ -1,45 +1,35 @@
...
 }
 
 export function initRepoArchiveLinks() {
-  $('.archive-link').on('click', function (event) {
-    event.preventDefault();
-    const url = this.getAttribute('href');
-    if (!url) return;
-    getArchive($(event.target), url, true);
-  });
+  queryElems('a.archive-link[href]', (el) => el.addEventListener('click', onDownloadArchive));
```
Comment (silverwind) [inline web_src/js/features/repo-common.js]:
```suggestion
  for (const el of document.querySelectorAll('a.archive-link[href]') {
    el.addEventListener('click', onDownloadArchive);
  }
```

File: web_src/js/utils/dom.js [hunk 2]
```diff
@@ -55,6 +55,10 @@ export function queryElemSiblings(el, selector) {
   return Array.from(el.parentNode.children).filter((child) => child !== el && child.matches(selector));
 }
 
+export function queryElems(selector, fn) {
+  return applyElemsCallback(document.querySelectorAll(selector), fn);
+}
+
```
Comment (silverwind) [inline web_src/js/utils/dom.js]:
```suggestion
```

**Comment by silverwind** [comment]
I replaced `applyElemsCallback` with `querySelectorAll`, will test later.

**Comment by silverwind** [comment]
Fixed and tested

**Review by silverwind — APPROVED** [review]

**Review by delvh — APPROVED** [review]

**Pull request merged**